File: app/src/main/java/com/wang/javamvp/base/rx/errorbean/ApiException.java
package com.wang.javamvp.base.rx.errorbean;

/**
 * @author Mis Wang
 * @date 2018/5/14  13:59
 * @fuction
 */
public class ApiException extends RuntimeException {
    private String code;
    private String errorMsg;

    public ApiException(String code, String msg) {
        super(msg);
        this.code = code;
        this.errorMsg = msg;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getErrorMsg() {
        return errorMsg;
    }

    public void setErrorMsg(String errorMsg) {
        this.errorMsg = errorMsg;
    }
}

File: app/src/main/java/com/wang/kotlinmvp/base/rx/errorbean/ApiException.kt
package com.wang.kotlinmvp.base.rx.errorbean

/**
 * @author Mis Wang
 * @date  2018/5/11  14:17
 * @fuction 运行时异常捕获
 */


/**
 * @param code 服务器返回的状态码
 * @param errorMsg 服务其返回的信息
 */
class ApiException(var code: String?, var errorMsg: String?) : RuntimeException(errorMsg)
